### PR TITLE
Upgrade versions of Scala, sbt and scalatest

### PR DESCRIPTION
**Pull request by mpociecha**
By the way I got rid of some new deprecation warnings.

### CODE REVIEW

File: src/test/scala/com/virtuslab/scalaworkshop/intro/IntroSpec.scala [hunk 1]
```diff
@@ -42,7 +42,7 @@ class IntroSpec extends WordSpec with Matchers {
         val value = throw new IllegalStateException()
         fail("Exception should have been thrown")
       } catch {
-        case ex => println("I am done")
+        case ex: Throwable => println("I am done")
```
Comment (odisseus) [inline src/test/scala/com/virtuslab/scalaworkshop/intro/IntroSpec.scala]: Consider using [`NonFatal`](https://www.scala-lang.org/api/current/scala/util/control/NonFatal$.html).

Comment (mpociecha) [reply src/test/scala/com/virtuslab/scalaworkshop/intro/IntroSpec.scala]: I know. But it's not my example and, I think, Paweł wanted to have this code as simple as possible (without having to explain what NonFatal means).

Comment (mpociecha) [reply src/test/scala/com/virtuslab/scalaworkshop/intro/IntroSpec.scala]: @pdolega Correct me if I'm wrong.

Comment (odisseus) [reply src/test/scala/com/virtuslab/scalaworkshop/intro/IntroSpec.scala]: Then just use `IllegalStateException`, as it doesn't introduce new entities. By the way, expecting  a broad variety of exceptions is IMO an antipattern.

Comment (pdolega) [reply src/test/scala/com/virtuslab/scalaworkshop/intro/IntroSpec.scala]: I'd say @odisseus is right. Can change it today in the evening.

Comment (mpociecha) [reply src/test/scala/com/virtuslab/scalaworkshop/intro/IntroSpec.scala]: I know all of this...

**Pull request merged**